### PR TITLE
BigData rename

### LINKED ISSUE #403: standardize API Entry or Element
also see #401

### PR DESCRIPTION
**Pull request by Affie**
- close #403 : rename Elements->Blob
- BigData -> Data

### CODE REVIEW

File: src/BigData/BigData.jl [hunk 1]
```diff
@@ -10,8 +10,9 @@ include("services/FileDataStore.jl")
 
 export AbstractDataStore
 
-export AbstractBigDataEntry, GeneralBigDataEntry, MongodbBigDataEntry, FileBigDataEntry
+export AbstractBigDataEntry, GeneralDataEntry, MongodbDataEntry, FileDataEntry
```
Comment (GearsAD) [inline src/BigData/BigData.jl:13]: Could you this change to AbstractDataEntry?

Comment (Affie) [reply src/BigData/BigData.jl:13]: Hi. Yes, definitely.  I changed it broke a bunch of stuff and then changed it back for the commit.

Comment (GearsAD) [reply src/BigData/BigData.jl:13]: :+1:

Comment (Affie) [reply src/BigData/BigData.jl:13]: Any suggestions on how to deprecate an Abstract type?

**Comment by Affie** [comment]
`global _invalidIds = ["USER", "ROBOT", "SESSION", "VARIABLE", "FACTOR", "ENVIRONMENT", "PPE", "BIGDATA", "FACTORGRAPH"]`

What to do with BIGDATA keyword here?

**Comment by Affie** [comment]
As per our discussion, I will leave this PR as only the name changes. Should implement UUID and improve tests.

**Comment by GearsAD** [comment]
> `global _invalidIds = ["USER", "ROBOT", "SESSION", "VARIABLE", "FACTOR", "ENVIRONMENT", "PPE", "BIGDATA", "FACTORGRAPH"]`
> 
> What to do with "BIGDATA" here? Also just "DATA" or "DATABLOB" or "DATA_ENTRY" - went with this one
> And should it be changed as part of CGDFG updates?

I guessed DATA in CGDFG and used that :+1:

**Comment by dehann** [comment]
sounds good to me, you guys should merge as soon as ready

File: src/BigData/services/AbstractBigDataEntries.jl [hunk 2]
```diff
@@ -125,24 +124,25 @@ Related
...
+                          dfglabel::Symbol,
+                          datastore::Union{FileDataStore, InMemoryDataStore},
+                          datalabel::Symbol)
   #
   vari = getVariable(dfg, dfglabel)
   if !hasDataEntry(vari, datalabel)
-    @error "missing data entry $datalabel in $dfglabel"
-    return nothing, nothing
+      # current standards is to fail hard
+    error("missing data entry $datalabel in $dfglabel")
```
Comment (GearsAD) [inline src/BigData/services/AbstractBigDataEntries.jl:135]: Gotcha, so fail loud :+1:

File: src/services/Serialization.jl [hunk 3]
```diff
@@ -16,8 +16,10 @@ function packVariable(dfg::G, v::DFGVariable)::Dict{String, Any} where G <: Abst
     props["smallData"] = JSON2.write(v.smallData)
     props["solvable"] = v.solvable
     props["softtype"] = string(typeof(getSofttype(v)))
-    props["bigData"] = JSON2.write(Dict(keys(v.bigData) .=> map(bde -> JSON2.write(bde), values(v.bigData))))
-    props["bigDataElemType"] = JSON2.write(Dict(keys(v.bigData) .=> map(bde -> typeof(bde), values(v.bigData))))
+    # props["bigData"] = JSON2.write(Dict(keys(v.dataDict) .=> map(bde -> JSON2.write(bde), values(v.dataDict))))
+    # props["bigDataElemType"] = JSON2.write(Dict(keys(v.dataDict) .=> map(bde -> typeof(bde), values(v.dataDict))))
+    props["dataEntry"] = JSON2.write(Dict(keys(v.dataDict) .=> map(bde -> JSON2.write(bde), values(v.dataDict))))
```
Comment (GearsAD) [inline src/services/Serialization.jl:21]: Definitely 0.9

File: test/DataStoreTests.jl [hunk 4]
```diff
@@ -1,52 +1,85 @@
-using DistributedFactorGraphs
+if false
```
Comment (GearsAD) [inline test/DataStoreTests.jl:1]: Is the plan to ignore tests until the next PR?

File: src/BigData/services/AbstractBigDataEntries.jl (hunk 2 above)
Comment (Affie) [reply src/BigData/services/AbstractBigDataEntries.jl:135]: Jip, as per #145. I would still like the option of not failing hard though. We can first standardize the behaviour and then add to it.

File: src/services/Serialization.jl (hunk 3 above)
Comment (Affie) [reply src/services/Serialization.jl:21]: I tried to make it backwards compatible. Still need to test it though.

File: test/DataStoreTests.jl (hunk 4 above)
Comment (Affie) [reply test/DataStoreTests.jl:1]: Yes, I ran into problems with tests. I wanted to get the name change in ASAP as it will influence you.

**Pull request merged**